File: 05-multiplication-table/mult_table.rb
# Ruby program to generate multiplication tables
# Created by M.V.Harish Kumar on 10-09-2023

def gen_tables(number, times)
    puts "#{number} Tables"
    (1..times).each do |i|
        puts "#{number} × #{i} = #{number*i}"
    end
end

print "Enter the no. to generate tables: "
n = gets.to_i
print "Enter the no. till which to generate: "
t = gets.to_i
gen_tables n, t

File: 05-multiplication-table/mult_table.exs
# Elixir program to generate multiplication tables
# Created by M.V.Harish Kumar on 22-12-2023

gen_tables = fn(number, times) ->
  IO.puts "#{number} Tables\n"
  Enum.each(1..times, fn i ->
    IO.puts "#{number} × #{i} = #{number*i}"
  end)
end

{n, _} = IO.gets("Enter the no. to generate tables: ") |> Integer.parse
{t, _} = IO.gets("Enter the no. till which to generate: ") |> Integer.parse
gen_tables.(n, t)
